client-py/casperlabs_client/vdag.py

```diff
@@ -140,6 +142,18 @@ def lane(validator, block_infos, min_rank, max_rank, genesis_block_id):
     )
 
 
+def call_dot(dot_dag_description, file_name, file_format):
+    with tempfile.NamedTemporaryFile(mode="w") as f:
+        f.write(dot_dag_description)
+        f.flush()
+        cmd = f"dot -T{file_format} -o {file_name} {f.name}"
+        rc = os.system(cmd)
+        if rc:
+            raise Exception(f"Call to dot ({cmd}) failed with error code {rc}")
```

Comment:
Please consider using `subprocess.call` as opposed to `os.system` as it uses a shell to execute command which can be harmful.

```suggestion
        subprocess.call(["dot", "-T", file_format, "-o", file_name, f.name])
```

Reply:
I will update this. Just moved the old code over.